tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -185,25 +500,17 @@ def __init__(self, target_executors):
     Raises:
       ValueError: If the value is unrecognized (e.g., a nonexistent intrinsic).
     """
-    py_typecheck.check_type(target_executors, dict)
+    py_typecheck.check_subclass(intrinsic_strategy, IntrinsicStrategy)
+    self.intrinsic_strategy = intrinsic_strategy(self)
```

Comment:
one problem with initializing the class inside the ctor is that users can't easily control this process. an alternative might be to simply type check the strategy as being callable so that users may pass in eg a partial function.

Reply:
I would probably need to stick to your alternative suggestion, since we need access to the FederatingExecutor instance for its method execution. This also forces the strategy to use the executor placements passed to the FederatingExecutor instance, which seems useful.

Reply:
changed the input parameter to `intrinsic_strategy_fn` which is expected to be a callable producing an `IntrinsicStrategy` implementation